Index: linux-4.11.0-44.2.1.el7a.ppc64le/drivers/net/ethernet/broadcom/tg3.c
===================================================================
--- linux-4.11.0-44.2.1.el7a.ppc64le.orig/drivers/net/ethernet/broadcom/tg3.c
+++ linux-4.11.0-44.2.1.el7a.ppc64le/drivers/net/ethernet/broadcom/tg3.c
@@ -7453,10 +7453,6 @@ static irqreturn_t tg3_msi_1shot(int irq
 	struct tg3_napi *tnapi = dev_id;
 	struct tg3 *tp = tnapi->tp;
 
-	prefetch(tnapi->hw_status);
-	if (tnapi->rx_rcb)
-		prefetch(&tnapi->rx_rcb[tnapi->rx_rcb_ptr]);
-
 	if (likely(!tg3_irq_sync(tp)))
 		napi_schedule(&tnapi->napi);
 
